openlibrary/plugins/importapi/code.py

```diff
@@ -348,14 +354,32 @@ def get_ia_record(metadata: dict) -> dict:
             d['description'] = description
         if isbn:
             d['isbn'] = isbn
-        if language and len(language) == 3:
-            d['languages'] = [language]
+        if language:
+            if len(language) == 3:
+                d['languages'] = [language]
+
+            # Try converting the name of a language to its three character code.
+            # E.g. English -> eng.
+            else:
+                try:
+                    if lang_code := get_abbrev_from_full_lang_name(language):
+                        d['languages'] = [lang_code]
+                except LanguageMatchError as e:
+                    logger.warning(
+                        "Multiple language matches for %s. No edition language set for %s.",
+                        e.language_name,
+                        metadata.get("identifier"),
+                    )
+
         if lccn:
             d['lccn'] = [lccn]
         if subject:
             d['subjects'] = subject
         if oclc:
             d['oclc'] = oclc
+        # Ensure no negative page number counts.
+        if number_of_pages and int(number_of_pages) - 4 >= 1:
+            d['number_of_pages'] = int(number_of_pages) - 4
```

Comment:
we still want to set it even if imagecount is small; just set it to the unsubtracted imagecount

Reply:
I made this change.